snapcraft/tests/sources/test_7z.py

```diff
@@ -0,0 +1,62 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2017 Tim Süberkrüb
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import os
+import os.path
+import shutil
+
+from snapcraft.internal import sources
+from snapcraft import tests
+
+
+class Test7z(tests.TestCase):
+
+    @staticmethod
+    def relative_path():
+        return os.path.dirname(os.path.realpath(__file__))
+
+    def test_pull_7z_file_must_extract(self):
+        test_7z_file_path = os.path.join(
+            Test7z.relative_path(),
+            '7z', 'test.7z'
```

Comment:
Ideally, the unit test will build the test.7z archive instead of using a binary. 
What are our options here?